source/Playnite/Controllers/GenericGameController.cs

```diff
@@ -133,5 +145,59 @@ private void Monitor_TreeDestroyed(object sender, EventArgs args)
             stopWatch.Stop();
             OnStopped(this, new GameControllerEventArgs(this, stopWatch.Elapsed.TotalSeconds));
         }
+
+        private void CheckGameImagePath()
+        {
+            if (Game.PlayAction.Type == GameActionType.Emulator)
+            {
+                var gameImagePath = Game.GameImagePath;
+                if (!FileSystem.CheckDrivesForValidFilePath(ref gameImagePath, false))
+                {
+                    throw new FileNotFoundException($"File \"{gameImagePath}\" does not exist", gameImagePath);
+                }
+                else if (!Game.GameImagePath.Equals(gameImagePath, StringComparison.OrdinalIgnoreCase))
+                {
+                    logger.Warn($"ROM/Image \"{Game.GameImagePath}\" does not exist for game \"{Game.Name}\"" +
+                        $" and is temporarily changed to {gameImagePath}");
+                    Game.GameImagePath = gameImagePath;
+                }
+            }
+        }
+
+        private void CheckGameActionPath(GameAction gameAction)
+        {
+            if (Game.PlayAction.Type == GameActionType.File)
+            {
+                var gameActionPath = gameAction.Path;
+                if (!FileSystem.CheckDrivesForValidFilePath(ref gameActionPath, false))
+                {
+                    throw new FileNotFoundException($"File \"{gameActionPath}\" does not exist", gameActionPath);
+                }
+                else if (!gameAction.Path.Equals(gameActionPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    logger.Warn($"Path \"{gameAction.Path}\" does not exist for game \"{Game.Name}\"" +
+                        $" and is temporarily changed to {gameActionPath}");
+                    gameAction.Path = gameActionPath;
+                }
+            }
+        }
+
+        private void CheckEmulatorConfigExecutable(EmulatorProfile emulatorProfile)
+        {
+            if (Game.PlayAction.Type == GameActionType.Emulator)

```

Comment:
Working directory has to be handled as well. Some Play actions have relative Path, but use absolute WorkingDir instead. The same goes for emulators.